cdap-unit-test/src/test/java/co/cask/cdap/spark/SparkTestRun.java

```diff
@@ -294,6 +302,50 @@ public void testScalaSparkWithObjectStore() throws Exception {
   }
 
   @Test
+  public void testScalaSparkCrossNSStream() throws Exception {
```

Comment:
so you have tests for cross namespace streams in scala, and cross namespace datasets in java. Will it be simple to have all combinations?